### ISSUE TITLE
Add flag for functional.Jacobian to return output as well

### ISSUE THREAD

**Issue by hbendekgey** [pytorch/pytorch#70870]
### 🚀 The feature, motivation and pitch

I am working on implicit differentiation of an expensive function. I would like both its output (with computational graph intact) and the Jacobian of the function with respect to one of its inputs. This can be easily done by running the function twice, but I would prefer to save the computation time.

For example, I want to be able to run:

```
input = torch.randn(5)
param = torch.randn(5).requires_grad_()
def func(input):
    return input**2 + param

output, J = torch.autograd.functional.jacobian(func, input)
```

Such that output.backward() propagates gradients to `param` (whether or not it is connected to input in the computation graph is not important to me).

I have implemented this by a one-line change to the jacobian function to also return the output as well. I don't know if this is a use-case other people have, enough that it would warrant adding a "return_output" flag to the Jacobian function. At a high level, it feels odd to me that there is no built-in way to get an output which is computed as an intermediate step of the Jacobian calculation.

**Comment by soulitzer** [comment]
Currently, you can do this with functorch's jacrev. See the following example from https://pytorch.org/functorch/stable/generated/functorch.jacrev.html.

```python
from functorch import jacrev
x = torch.randn(5)

def f(x):
  return x.sin()

def g(x):
  result = f(x)
  return result, result

jacobian_f, f_x = jacrev(g, has_aux=True)(x)
assert torch.allclose(f_x, f(x))
```
